java-manta-client-unshaded/src/main/java/com/joyent/manta/exception/MantaErrorCode.java

```diff
@@ -180,4 +532,52 @@ public static MantaErrorCode valueOfCode(final Object object) {
 
         return valueOfCode(object.toString());
     }
+
+    /**
+     * Populates a Inclusive ArrayList for {@link MantaErrorCode} by its code value.
+     *
+     * @return a thread-safe Manta Errors List.
+     * @throws MantaClientException thrown if deprecated fields are not present or due to security violation.
+     */
+    public static List<MantaErrorCode> getNonDeprecatedList() throws MantaClientException {
+        nonDeprecatedList = Collections.synchronizedList(new ArrayList<>());
+        for (MantaErrorCode mantaErrorCode : values()) {
+            try {
+                Field field = MantaErrorCode.class.getField(mantaErrorCode.name());
+                if (field.isAnnotationPresent(Deprecated.class)) {
+                    continue;
+                }
+
+                nonDeprecatedList.add(mantaErrorCode);
+            } catch (NoSuchFieldException | SecurityException e) {
+                throw new MantaClientException("Security Violation detected "
+                        + "Rest Code not marked deprecated", e);
+            }
+        }
+
+        return nonDeprecatedList;
+    }
+
+    /**
+     * Populates an Inclusive Error Map for {@link MantaErrorCode} by its code value.
+     *
+     * @return a thread-safe Manta Errors HashMap.
+     * @throws MantaClientException if a thread-safe access is not granted for Error Map.
+     */
+    public static AtomicReference<Map<String, MantaErrorCode>> createErrorMap() throws MantaClientException {
```

Comment:
You don't need to use `AtomicReference` when you instantiate in the static scope eagerly.

Reply:
Noted.